### PR TITLE
Dto enhancement

### PR DESCRIPTION
**Pull request by supragyan**
Did development of DTO Enhancement.
1.Created JQGridJSON page for displaying Product table data in JSON format.
2.Created JSONService.cs for selecting data from Product table.
3.Created and implemented code for SaveJQGridJSON method.

### CODE REVIEW

File: root/programs/C#/Samples/WebApp_sample/ProjectX_sample/Aspx/testFxLayerP/table/testJQGridJson.aspx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    <link href="../../../Content/jquery.jqGrid/ui.jqgrid.css" rel="stylesheet" type="text/css" />
+    <script src="../../../Scripts/jquery-2.1.3.min.js" type="text/javascript"></script>
+    <script src="../../../Scripts/jquery-ui-1.11.2.min.js" type="text/javascript"></script>
+    <script src="../../../Scripts/i18n/grid.locale-ja.js" type="text/javascript"></script>
+    <script src="../../../Scripts/jquery.jqGrid.min.js" type="text/javascript"></script>
+    <script type="text/javascript">
+        $(function () {
+            $('#list').jqGrid({
+                datatype: 'json',
+                url: 'JQGridHandler.ashx',
```
Comment (yusukemaegawa) [inline root/programs/C#/Samples/WebApp_sample/ProjectX_sample/Aspx/testFxLayerP/table/testJQGridJson.aspx]: File 'JQGridHandler.ashx' does not exist.
You should describe WCF service name (*.svc) here.
(An old cord is left?)

File: root/programs/C#/Samples/WebApp_sample/ProjectX_sample/Aspx/testFxLayerP/table/testJQGridJson.aspx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+                ],
+                rowNum: 1000,
+                loadonce: false,
+                caption: 'JSON Sample (ページングなし)'
+            });
+
+            $('#list2').jqGrid({
+                datatype: 'json',
+                url: 'JQGridHandler.ashx',
```
Comment (yusukemaegawa) [inline root/programs/C#/Samples/WebApp_sample/ProjectX_sample/Aspx/testFxLayerP/table/testJQGridJson.aspx]: File 'JQGridHandler.ashx' does not exist.
You should describe WCF service name (*.svc) here.
(An old cord is left?)

File: root/programs/C#/Samples/WebApp_sample/ProjectX_sample/App_Code/JSONService.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    /// <summary>
+    /// GetProductData method for fetching product table data.
+    /// </summary>
+    public void GetProductData()
+    {
+        HttpContext.Current.Session["DAP"] = "SQL";
+        HttpContext.Current.Session["DBMS"] = DbEnum.DBMSType.SQLServer;
+
+        ProductsTableAdapter productTableAdapter = new ProductsTableAdapter();
+        DataTable productTableData = productTableAdapter.SelectMethod(0, 30);
```
Comment (yusukemaegawa) [inline root/programs/C#/Samples/WebApp_sample/ProjectX_sample/App_Code/JSONService.cs]: You must acquire 'startRowIndex', 'maximumRows' from a value of QueryString.
(you must not substitute a fixed value such as "0" and "30")

File: root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTable.cs [hunk 4]
```diff
@@ -400,5 +407,154 @@ public void AcceptChanges()
...
+        #region GetJson
+
+        /// <summary>
+        /// GetJson method for selecting data
+        /// </summary>
+        /// <returns></returns>
+        public JQGridDataClass GetJson()
+        {
+            // サンプルデータ取得
+            System.Data.DataTable dt = this.CreateSampleData();
```
Comment (yusukemaegawa) [inline root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTable.cs]: We asked you not to use sample data before.
However, the code is not revised according to the request.
(An old code is left?)

File: root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTable.cs [hunk 5]
```diff
@@ -400,5 +407,154 @@ public void AcceptChanges()
...
+
+        #endregion
+
+        #region CreateSampleData
+
+        /// <summary>
+        /// Creates Sample Data in datatable.
+        /// </summary>
+        /// <returns></returns>
+        private System.Data.DataTable CreateSampleData()
```
Comment (yusukemaegawa) [inline root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTable.cs]: We asked you not to use sample data before.
However, the code is not revised according to the request.
(An old code is left?)

File: root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTables.cs [hunk 6]
```diff
@@ -65,7 +73,7 @@ public class TableRecords
         #region インスタンス変数
 
         /// <summary>表を保持するList</summary>
-        private List<DTTable> _tbls = new List<DTTable>();
+        public List<DTTable> _tbls = new List<DTTable>();
```
Comment (yusukemaegawa) [inline root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTables.cs]: You can acquire a table of DTTables using Indexer.
We think that it is not necessary to declare variable '_tbls' as 'public'.
(we did similar indication before)

File: root/programs/C#/Samples/WebApp_sample/ProjectX_sample/App_Code/JSONService.cs (hunk 3 above)
Comment (yusukemaegawa) [inline root/programs/C#/Samples/WebApp_sample/ProjectX_sample/App_Code/JSONService.cs]: Similarly, you must acquire 'SortExpression', 'SortDirection' from QueryString.

File: root/programs/C#/Samples/WebApp_sample/ProjectX_sample/App_Code/JSONService.cs [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+        int count = productTableAdapter.SelectCountMethod();
+
+        //Calling SavejqGridJson
+        DTTable dtTable = new DTTable("Product");
+        dtTable.SavejqGridJson(productTableData, count);
```
Comment (yusukemaegawa) [inline root/programs/C#/Samples/WebApp_sample/ProjectX_sample/App_Code/JSONService.cs]: The 'DTTable' class does not have 'SavejqGridJson' method.
(a code of 'DTTable.cs' is old?)

File: root/programs/C#/Samples/WebApp_sample/ProjectX_sample/App_Code/IJSONService.cs [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+using System.Text;
+using System.ServiceModel.Web;
+
+// NOTE: You can use the "Rename" command on the "Refactor" menu to change the interface name "IJSONService" in both code and config file together.
+[ServiceContract(SessionMode = SessionMode.Allowed)]
+public interface IJSONService
+{
+    [OperationContract]
+    [WebGet(ResponseFormat = WebMessageFormat.Json, UriTemplate = "GetProductData")]
+    void GetProductData();
```
Comment (yusukemaegawa) [inline root/programs/C#/Samples/WebApp_sample/ProjectX_sample/App_Code/IJSONService.cs]: This WCF service must return JSON data to show jqGrid.
Therefore, the type of the return value must not be 'void'.

File: root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTable.cs (hunk 4 above)
Comment (kishoreandukuri) [inline root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTable.cs]: yes this is old code. New code didn't commit

File: root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTable.cs (hunk 5 above)
Comment (kishoreandukuri) [inline root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTable.cs]: yes this is old code. New code didn't commit

File: root/programs/C#/Samples/WebApp_sample/ProjectX_sample/App_Code/JSONService.cs (hunk 7 above)
Comment (kishoreandukuri) [inline root/programs/C#/Samples/WebApp_sample/ProjectX_sample/App_Code/JSONService.cs]: yes this is old code. New code didn't commit

File: root/programs/C#/Samples/WebApp_sample/ProjectX_sample/Aspx/testFxLayerP/table/testJQGridJson.aspx (hunk 1 above)
Comment (kishoreandukuri) [inline root/programs/C#/Samples/WebApp_sample/ProjectX_sample/Aspx/testFxLayerP/table/testJQGridJson.aspx]: yes this is old code. New code didn't commit

File: root/programs/C#/Samples/WebApp_sample/ProjectX_sample/Aspx/testFxLayerP/table/testJQGridJson.aspx (hunk 2 above)
Comment (kishoreandukuri) [inline root/programs/C#/Samples/WebApp_sample/ProjectX_sample/Aspx/testFxLayerP/table/testJQGridJson.aspx]: yes this is old code. New code didn't commit

File: root/programs/C#/Samples/WebApp_sample/ProjectX_sample/Aspx/testFxLayerP/table/testJQGridJson.aspx (hunk 1 above)
Comment (yusukemaegawa) [inline root/programs/C#/Samples/WebApp_sample/ProjectX_sample/Aspx/testFxLayerP/table/testJQGridJson.aspx]: In the condition that an old code is remaining, we cannot confirm whether the code that you wrote works correctly.
In such a half-finished state, we cannot merge your pull request.
Including a new code, please commit again.

File: root/programs/C#/Samples/WebApp_sample/ProjectX_sample/Aspx/testFxLayerP/table/testJQGridJson.aspx (hunk 2 above)
Comment (yusukemaegawa) [inline root/programs/C#/Samples/WebApp_sample/ProjectX_sample/Aspx/testFxLayerP/table/testJQGridJson.aspx]: In the condition that an old code is remaining, we cannot confirm whether the code that you wrote works correctly.
In such a half-finished state, we cannot merge your pull request.
Including a new code, please commit again.

File: root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTable.cs (hunk 4 above)
Comment (yusukemaegawa) [inline root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTable.cs]: In the condition that an old code is remaining, we cannot confirm whether the code that you wrote works correctly.
In such a half-finished state, we cannot merge your pull request.
Including a new code, please commit again.

File: root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTable.cs (hunk 5 above)
Comment (yusukemaegawa) [inline root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTable.cs]: In the condition that an old code is remaining, we cannot confirm whether the code that you wrote works correctly.
In such a half-finished state, we cannot merge your pull request.
Including a new code, please commit again.

File: root/programs/C#/Samples/WebApp_sample/ProjectX_sample/App_Code/JSONService.cs (hunk 7 above)
Comment (yusukemaegawa) [inline root/programs/C#/Samples/WebApp_sample/ProjectX_sample/App_Code/JSONService.cs]: In the condition that an old code is remaining, we cannot confirm whether the code that you wrote works correctly.
In such a half-finished state, we cannot merge your pull request.
Including a new code, please commit again.

File: root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTable.cs (hunk 5 above)
Comment (kishoreandukuri) [inline root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTable.cs]: HODC will commit the new code changes after finishing review comment changes.

**Pull request closed without merging**